src/python/CRABClient/Commands/setdatasetstatus.py

```diff
@@ -0,0 +1,118 @@
+        status = self.options.status
+        recursive = self.options.recursive
+        self.logger.debug('instance  = %s' % instance)
+        self.logger.debug('dataset   = %s' % dataset)
+        self.logger.debug('status    = %s' % status)
+        self.logger.debug('recursive = %s' % recursive)
+
+        if recursive:
+            self.logger.warning("ATTENTION: recursive option is not implemented yet. Ignoring it")
+
+        # from DBS instance, to DBS REST services
+        dbsReader, dbsWriter = getDbsREST(instance=instance, logger=self.logger,
+                                          cert=self.proxyfilename, key=self.proxyfilename)
+
+        self.logger.info("looking up Dataset %s in DBS %s" % (dataset, instance))
+        datasetStatusQuery = {'dataset': dataset, 'dataset_access_type': '*', 'detail': True}
+        ds, rc, msg = dbsReader.get(uri="datasets", data=urlencode(datasetStatusQuery))
+        self.logger.debug('exitcode= %s', rc)
+        if not ds:
+            self.logger.error("ERROR: dataset %s not found in DBS" % dataset)
+            raise ConfigurationException
+        self.logger.info("Dataset status in DBS is %s" % ds[0]['dataset_access_type'])
+        self.logger.info("Will set it to %s" % status)
+        data = {'dataset': dataset, 'dataset_access_type': status}
+        jdata = json.dumps(data)
+        out, rc, msg = dbsWriter.put(uri='datasets', data=jdata)
+        if rc == 200 and msg == 'OK':
+            self.logger.info("Dataset status changed successfully")
+            result = 'SUCCESS'
+        else:
+            msg = "Dataset status change failed: %s" % out
+            raise CommandFailedException(msg)
+
+        ds, rc, msg = dbsReader.get(uri="datasets", data=urlencode(datasetStatusQuery))
+        self.logger.debug('exitcode= %s', rc)
+        self.logger.info("Dataset status in DBS now is %s" % ds[0]['dataset_access_type'])
+
+        return {'commandStatus': result}
+
+    def setOptions(self):
+        """
+        __setOptions__
+
+        This allows to set specific command options
+        """
+        self.parser.add_option('--instance', dest='instance', default='prod/phys03',
+                               help="DBS instance. e.g. prod/phys03 (default) or int/phys03. Use at your own risk." + \
+                                    "Unless you really know what you are doing, stay with the default"
+                               )
+        self.parser.add_option('--dataset', dest='dataset', default=None,
+                               help='dataset name')
+        self.parser.add_option('--status', dest='status', default=None,
+                               help="New status of the dataset: VALID/INVALID/DELETED/DEPRECATED",
+                               choices=['VALID', 'INVALID', 'DELETED', 'DEPRECATED']
+                               )
+        self.parser.add_option('--recursive', dest='recursive', default=False, action="store_true",
```

Comment:
just to make sure i understand correctly, this applies to children datasets and their files, but not to the files of the current dataset that is being invalidated, doesn't it?

Reply:
the idea (from the old DBS script) is to also invalidate all files in current dataset, no matter if --recursive is there or not :man_shrugging: